src/NuGet.Frameworks/DefaultFrameworkMappings.cs

```diff
@@ -332,6 +332,54 @@ public IEnumerable<OneWayCompatibilityMappingEntry> CompatibilityMappings
                                 new FrameworkRange(
                                     new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.WinRT, FrameworkConstants.EmptyVersion),
                                     new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.WinRT, new Version(4, 5, 0, 0)))),
+
+                            // Dotnet compatibility white list
+                            // net45 and up support dotnet
+                            new OneWayCompatibilityMappingEntry(
+                                new FrameworkRange(
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Net, new Version(4, 5, 0, 0)),
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Net, FrameworkConstants.MaxVersion)),
+                                FrameworkConstants.DotNetAll),
+
+                            // dnx451 and up support dotnet
+                            new OneWayCompatibilityMappingEntry(
+                                new FrameworkRange(
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Dnx, new Version(4, 5, 1, 0)),
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Dnx, FrameworkConstants.MaxVersion)),
+                                FrameworkConstants.DotNetAll),
+
+                            // silverlight above version 5 supports dotnet
+                            new OneWayCompatibilityMappingEntry(
+                                new FrameworkRange(
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Silverlight, FrameworkConstants.Version5),
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Silverlight, FrameworkConstants.MaxVersion),
+                                    includeMin: false,
+                                    includeMax: true),
+                                FrameworkConstants.DotNetAll),
+
+                            // wp8 and up support dotnet
+                            new OneWayCompatibilityMappingEntry(
+                                new FrameworkRange(
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.WindowsPhone, new Version(8, 0, 0, 0)),
+                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.WindowsPhone, FrameworkConstants.MaxVersion)),
+                                FrameworkConstants.DotNetAll),
```

Comment:
Let's make this `dotnet50` for now. Things are still up-in-the-air a bit and it'd be better to be conservative for now.